### PR TITLE
update schema and add formulas to table config

### PR DESCRIPTION
**Pull request by hillalex**
This PR:
* adds `precision` as an optional field in table config schema
* adds the real cost algorithms to table config with tests

MINT changes to support this: https://github.com/mrc-ide/mint/pull/65

### CODE REVIEW

File: inst/json/table_data.json [hunk 1]
```diff
@@ -3,10 +3,10 @@
     "intervention": "none",
     "netUse": 0,
     "irsUse": 0,
-    "casesAverted": 400,
-    "casesAvertedPer1000": 300,
-    "casesAvertedPer1000ErrorPlus": 16,
-    "casesAvertedPer1000ErrorMinus": 41,
+    "casesAverted": 0,
```
Comment (hillalex) [inline inst/json/table_data.json:6]: this is all still fake data, just put some zeros in for no intervention to mimic real data

File: tests/testthat/test-config.R [hunk 2]
```diff
@@ -47,6 +48,19 @@ get_expected_costs <- function() {
        costs_N2_S1 = costs_N2_S1)
 }
 
+get_incremental_increase_in_costs <- function() {
+  total_costs <- get_expected_total_costs()
+  baseline_cost <- as.list(rep(total_costs$costs_N0, 6))
+  increase_in_costs <- mapply("-", total_costs, baseline_cost, SIMPLIFY = FALSE)
```
Comment (richfitz) [inline tests/testthat/test-config.R:54]: You may like `Map` which does this behaviour by default (not simplifying)

File: tests/testthat/test-config.R [hunk 3]
```diff
@@ -171,7 +204,66 @@ test_that("prevalence graph config containe valid intervention ids", {
   IRS <- ids[[4]]
   expect_equal(IRS, mint_intervention(0, 1, "pto"))
   ITN_IRS <- ids[[5]]
-  expect_equal(ITN_IRS,mint_intervention(1, 1, "std"))
+  expect_equal(ITN_IRS, mint_intervention(1, 1, "std"))
```
Comment (richfitz) [inline tests/testthat/test-config.R:207]: Opened mrc-1975 to set up linting to help with this churn

**Review by richfitz — APPROVED** [review]

File: tests/testthat/test-config.R (hunk 3 above)
Comment (hillalex) [reply tests/testthat/test-config.R:207]: nice, thanks

**Pull request merged**